addons/hr_holidays/models/hr_leave.py

```diff
@@ -717,34 +758,34 @@ def add_follower(self, employee_id):
         if employee.user_id:
             self.message_subscribe(partner_ids=employee.user_id.partner_id.ids)
 
-    @api.constrains('holiday_status_id', 'date_to', 'date_from')
+    @api.constrains('holiday_allocation_id', 'date_to', 'date_from')
     def _check_leave_type_validity(self):
         for leave in self:
-            vstart = leave.holiday_status_id.validity_start
-            vstop = leave.holiday_status_id.validity_stop
+            vstart = leave.holiday_allocation_id.date_from
+            vstop = leave.holiday_allocation_id.date_to
             dfrom = leave.date_from
             dto = leave.date_to
-            if leave.holiday_status_id.validity_start and leave.holiday_status_id.validity_stop:
+            if leave.holiday_allocation_id.date_from and leave.holiday_allocation_id.date_to:
                 if dfrom and dto and (dfrom.date() < vstart or dto.date() > vstop):
                     raise ValidationError(_(
                         '%(leave_type)s are only valid between %(start)s and %(end)s',
                         leave_type=leave.holiday_status_id.display_name,
-                        start=leave.holiday_status_id.validity_start,
-                        end=leave.holiday_status_id.validity_stop
+                        start=leave.holiday_allocation_id.date_from,
+                        end=leave.holiday_allocation_id.date_to
                     ))
-            elif leave.holiday_status_id.validity_start:
+            elif leave.holiday_allocation_id.date_from:
                 if dfrom and (dfrom.date() < vstart):
                     raise ValidationError(_(
                         '%(leave_type)s are only valid starting from %(date)s',
                         leave_type=leave.holiday_status_id.display_name,
-                        date=leave.holiday_status_id.validity_start
+                        date=leave.holiday_allocation_id.date_from
                     ))
-            elif leave.holiday_status_id.validity_stop:
+            elif leave.holiday_allocation_id.date_to:
                 if dto and (dto.date() > vstop):
                     raise ValidationError(_(
                         '%(leave_type)s are only valid until %(date)s',
                         leave_type=leave.holiday_status_id.display_name,
-                        date=leave.holiday_status_id.validity_stop
+                        date=leave.holiday_allocation_id.date_to
```

Comment:
I think we could use vstart and vstop variables instead of holiday_allocation_id.date_from and holiday_allocation_id.date_to.